### ISSUE TITLE
docker-host cgroups v2 incompatiblity

### ISSUE THREAD

**Issue by smlx** [uselagoon/lagoon#2458]
# Describe the bug

Lagoon's `docker-host` is incompatible with docker using cgroups v2.

## To Reproduce

1. Upgrade to docker 20.10.2 on Linux.
2. Ensure that it is using cgroups v2 by running `docker info` and looking for: 
```
 Cgroup Driver: systemd
 Cgroup Version: 2
```
3. Run the Lagoon test suite.
4. See error during image build:

```
cgroups: cgroup mountpoint does not exist: unknown
```

You can also minimally get the error by executing this command inside the docker-host container:
```
docker run --rm -it --entrypoint sh  alpine:3.11
```

## Expected behavior

Docker dind should work. It does using the [20-dind tag on the docker image](https://hub.docker.com/_/docker).

## Screenshots

n/a

## Additional context

Maybe we should upgrade the docker dind base image we build docker-host on top of. However I note that in https://github.com/amazeeio/lagoon/commit/e152aee05cbcc401e60f8e13139ab912a9dd6e56 we explicitly pin the version due to the base alpine version? Any thoughts @tobybellwood?

I don't know if the `docker:20-dind` image is compatible with earlier docker host versions running on cgroups v1. Could someone check that?

**Comment by tobybellwood** [comment]
The only reason the image was pinned was because alpine 3.11 - I've no issues with moving it forwards - will have a look next week!

**Comment by smlx** [comment]
I tried bumping the base image of Lagoon's `docker-host` to use `docker:20-dind` and it _did_ fix the issue!

However.

I then hit [a bug](https://github.com/containerd/containerd/issues/4837) in `containerd` which is [fixed in git](https://github.com/containerd/containerd/pull/4845) but hasn't made it into a `containerd` release yet, much less a `kind` node image release. :cry: 

Unfortunately I think all we can do is wait for the fix to make its way into a `kind` node image.

**Comment by smlx** [comment]
This potentially has implications on bumping the `docker-host` version in general since we need to ensure that the k8s cluster `docker-host` runs on isn't affected by that containerd bug.

**Comment by tobybellwood** [comment]
fixed in #2705
